packages/notifications/push/amplify_push_notifications/ios/Classes/AmplifyPushNotificationsPlugin+AppDelegate.swift

```diff
@@ -0,0 +1,82 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Foundation
+import Flutter
+
+private let completionHandlerIdKey = "completionHandlerId"
+
+extension AmplifyPushNotificationsPlugin {
+    public func application(
+        _ application: UIApplication,
+        didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
+    ) {
+        let token = deviceToken.map { String(format: "%02.2hhx", $0) }.joined()
+        sharedEventsStreamHandlers.tokenReceived.sendEvent(payload: ["token": token])
+    }
+
+    public func application(
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions launchOptions: [AnyHashable : Any] = [:]
+    ) -> Bool {
+        if let remoteNotification = launchOptions[UIApplication.LaunchOptionsKey.remoteNotification] as? [AnyHashable: Any] {
+            // If the App is awaken by a remote notification when remote notifications background mode is enabled
+            if application.applicationState == .background {
+                awakenInBackground = true
+            }
+
+            // When the App is launched (from terminated state) by tapping on a notification
+            // This happens when remote notifications background mode is NOT enabled
+            if application.applicationState != .background {
```

Comment:
This should be an `else` clause since it's orthogonal to the `if` above and is the reason we don't set `launchNotification`